include/sys/fs/zfs.h

```diff
@@ -36,6 +36,7 @@
 
 #include <sys/time.h>
 #include <sys/zio_priority.h>
+#include <sys/zfs_context_os.h>
```

Comment:
It would be best if we didn't need to include `sys/zfs_context_os.h` from `sys/zfs.h`.  This adds another dependency and potentially drags in a lot of unnecessary headers.  I know we've been systematically moving these kind of thing in to platform specific headers.  But since this is only used to set the default `spa_config_path` I think it would be entirely reasonable to do this.  There are no user space consumers I'm aware of.


```c
/*
 * This can be overridden in userland to preserve an alternate namespace for
 * userland pools when doing testing.
 */
#if defined(__FreeBSD__)
char *spa_config_path = "/boot/zpool.cache";
#else
char *spa_config_path = "/etc/zfs/zpool.cache";
#endif
```